app/src/main/java/de/danoeh/antennapod/menuhandler/FeedItemMenuHandler.java

```diff
@@ -243,20 +235,9 @@ public static boolean onMenuItemClicked(@NonNull Fragment fragment, int menuItem
             case R.id.visit_website_item:
                 IntentUtils.openInBrowser(context, FeedItemUtil.getLinkWithFallback(selectedItem));
                 break;
-            case R.id.share_link_item:
-                ShareUtils.shareFeedItemLink(context, selectedItem);
-                break;
-            case R.id.share_download_url_item:
-                ShareUtils.shareFeedItemDownloadLink(context, selectedItem);
-                break;
-            case R.id.share_link_with_position_item:
-                ShareUtils.shareFeedItemLink(context, selectedItem, true);
-                break;
-            case R.id.share_download_url_with_position_item:
-                ShareUtils.shareFeedItemDownloadLink(context, selectedItem, true);
-                break;
-            case R.id.share_file:
-                ShareUtils.shareFeedItemFile(context, selectedItem.getMedia());
+            case R.id.share_item:
+                new ShareDialog(context, selectedItem)
+                        .show(((AppCompatActivity) context).getSupportFragmentManager(), "ShareEpisodeDialog");
```

Comment:
Please change the type of `context` instead of casting (`fragment.getActivity()`?). I fear that this could lead to runtime crashes otherwise.

Reply:
thanks, update it. For the next commit